### PR TITLE
MGMT-17230: Use IDMS when possible, take two

### PR DESCRIPTION
**Pull request by jhernand**
Currently when we run the 'oc' command in disconnected environments we generate a `ImageContentSourcePolicy` object inside a YAML file and reference it with the `--icsp-file` command line flag. That is deprecated since OpenShift 4.14, and replaced by the `ImageDigestMirrorSet` object and the `--idms-file` command line flag.

This patch changes the code so that it checks if the `--idms-file` command line flag is available, and if it is it generates a `ImageDigestMirrorSet` and uses the `--idms-file` flag instead of the deprecated `--icsp-file`.

The previous behavior is preserved when the `--idms-file` flag isn't available in order to keep supporting older OpenShift versions.

Note that this is the second take of this fix. The first was reverted because, unlike the other commands, the `oc image info` hasn't been updated to support the `--idms-file` flag, and we were running `oc adm release info --help` to detect that flag, regardless of the actual command that was then executed. We have changed the approach now so that if the command to be executed is `oc x` then we will run `oc x --help` to detect the flags. The unit tests have also been improved to check that the right flag is used for all the commands that we run.

Related: https://issues.redhat.com/browse/MGMT-17230
Related: https://github.com/openshift/assisted-service/pull/6764
Related: https://github.com/openshift/assisted-service/pull/6810
Related: https://github.com/openshift/enhancements/blob/master/enhancements/api-review/add-new-CRD-ImageDigestMirrorSet-and-ImageTagMirrorSet-to-config.openshift.io.md

- [ ] New Feature 
- [X] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [X] Automation (CI, tools, etc)
- [X] Cloud
- [X] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [X] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Checklist

- [X] Title and description added to both, commit and PR.
- [X] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [X] This change does not require a documentation update (docstring, `docs`, README, etc)
- [X] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by jhernand** [comment]
/hold

This pull request contains the revert, and then the new fix. Will need to be rebased once the revert is merged. I also want to test this fully in a disconnected environment.

**Comment by andfasano** [comment]
@jhernand you could also the the agent sno job, since it runs in a disconnected environment (and currently is impacted by this change, thanks @bfournie for the heads up).

/test e2e-agent-sno-ipv6

**Comment by jhernand** [comment]
> @jhernand you could also use the agent sno job, since it runs in a disconnected environment (and currently is impacted by this change, thanks @bfournie for the heads up).

I will use that job, thanks @andfasano . Note however that the issue is already fixed because we reverted the offending change (see #6764).

**Comment by jhernand** [comment]
@paul-maidment @CrystalChun in case you didn't notice, be aware that the first take on this broke installation in disconnected environments because I assumed that all the `oc ...` commands that we use have the same `--idms-file` option. That isn't the case, the `oc image info` command doesn't have it yet. This new version of the patch tries to address that. To do so it gets the available options from the command itself (previously it got them always from `oc adm release info`). I'd appreciate if you can review again.

**Review by CrystalChun — APPROVED** [review]
/lgtm

**Comment by CrystalChun** [comment]
Thanks @jhernand! Apologies I didn't help catch that `oc image info` didn't support `idms` flag in your initial PR either :(

**Pull request merged**
